dev-docs/RFCs/proposals/instance-culling-rfc.md

```diff
@@ -0,0 +1,178 @@
+
+- **distance culling** - Determining the distance at which each instance is anchored
+- **bounding boxes** - Using bounding boxes to automatically determine distance culling limits
+- **vertex filtering** - A simpler, less performant approach of filtering out unnecessary work by filtering vertices
+- **instance filtering** - A more powerful, potentially more performant approach of filtering out unnecessary rendering by filtering the instance list 
+- **run-time tiling** - Separating a big table in sub-tables based on octree or runtime KD-tree decomposition, in the style of I3S or 3D tiles. 
+- **LOD** - using different geometries
+- **Decimation** - Automatic generation of smaller, reduced detail geometries by implementing decimation algorithms.
+
+## Distance Culling
+
+The idea is to select which instances to draw based on the distance of their center points from the viewer. Many layers already have distance points.
+
+
+## Vertex Filtering
+
+A fairly low-cost way of reducing GPU work could be to apply a filter in the vertex shader that
+- checks the distance of the instance position from the camera
+- generates a degenerate 0 size polygon.
+- or sets a flag to discard the pixels in the vertex shader.
+
+This will likely generate some performance boost at a low cost but it is unlikely to provide order-of-magnitude improvement as all vertices for all instances are still processed by the vertex shader.
+
+## Instance filtering
+
+The other approach would be to filter the instance table based on the view, making a poass. 
+
+For non-instanced layers, it should be possible to just generate an index array
+that omits vertices belonging to filtered geometry.
+
+Unfortunately, GPUs do not provide indexing support for instance tables like they do for the primitive geometry, so we can't just provide an array with the instance indices we want to render, we need to generate a completely set of instance attributes.
+
+Generation of new instance arrays could be done on the CPU whenever the viewport changes, writing into new typed arrays and uploading those. 
+
+For layers that typically have modest amounts of instances (<10K instances) such as ScenegraphLayer the CPU approach is likely fairly performant as there is no GPU read back involved, and most likely will perform much better on large total vertex counts.
+
+For the general case, where we may deal with millions of instances, we could investigate if some transform feedback type technique could be used to filter the arrays on the GPU. Since the output is random access (not row-to-row) the results would presumably need to be written into a texture and converted to a buffer. Actual performance would have to be carefully measured to ensure that the desired performance characterstics are met.
+
+Note that gor filtering use cases, the performance criteria may be lower than rendering and the GPU technique might be more applicable.
+
+## Run-Time Tiling
```

Comment:
An issue with tiling is that it has to be regenerated when instance bounding box changes. This defeats the performance benefit of using e.g. `sizeScale` of the ScenegraphLayer.